### DISCUSSION TITLE
Allow provisional drafts to expire

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/cms#15354] [Ideas]
We often get complaints from clients that an entry form is not displaying the latest version of the entry that is published on the website. This is usually because they have a provisional draft that's been sitting there for a while, meanwhile other people have modified the entry, so the provisional draft is out of date. Of course this is a layer 8 problem, but it's frequent enough to think about ways of addressing this.

In my experience with my clients, this happens most often when clients accidentally create a provisional draft without noticing it. Then they don't expect to have pending changes, especially if it's been a while since they edited that entry, and don't notice the banner that informs them that they're seeing a provisional draft.

There are several ways to address this, including making the provisional draft notice more noticeable. However, one simple solution that might solve a lot of those problems would be an option in the general config to allow provisional drafts to expire after a set amount of time.

For example, one might set:

```php
new GeneralConfig::create()->provisionalDraftsExpirationDuration(604_800); 
```

This would expire provisional drafts after one week. Once a provisional draft expires, opening the entry edit form will no longer show the provisional draft. Instead you get the canonical entry and can start fresh. The provisional draft can be garbage collected. Of course, any change to a provisional draft should update its timestamp and reset the expiration timer.

By default, the config would be empty, so that provisional drafts don't expire (like the current behaviour). But this would allow the developer to define an appropriate expiration time based on the site's content and how frequently authors interact with the CMS.

**Comment by brandonkelly** [comment]
Are you seeing this on Craft 5.2+? It seems like it wouldn’t be an issue as much now that entries identify as changed within their chips and cards. (#14975)

**Reply by MoritzLost** [reply]
@brandonkelly Don't have clients actively using that version yet, no. Though I don't expect to see much change here. Yes, the element index now has the _Edited_ indicator, but that's even less noticeable than the banner at the top of the entry form. So I don't think it solves the problem.

Maybe there is no solution – if authors don't read the banner, maybe it's their own fault. The expiration date was just an idea for a stop-gap solution, because this has happened most often when clients have edited an entry a long time ago and didn't notice they still had a provisional draft. Though making the banner more noticeable or indicating the status in some other way might also help?

**Reply by brandonkelly** [reply]
Give them some time using 5.2 and let’s see how it shakes out. The “Edited” labels should be very noticeable as they are navigating to the entry, and they’re also displayed within the breadcrumbs. Lots of opportunities to understand what you’re looking at.

**Reply by MoritzLost** [reply]
@brandonkelly Thanks, we'll see how it goes.
